File: dto/dtos/CountryName.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using dto.Paging;

namespace dto.dtos
{
    public class CountryNameRes
    {
        public int CountryNameId { get; set; }

        public int? LanguageId { get; set; }
        public LanguageRes? Language { get; set; }

        public int? CountryId { get; set; }
        public CountryRes? Country { get; set; }

        public string? Name { get; set; }
    }
    public class CountryNameReqEdit
    {
        [Required]
        public int? LanguageId { get; set; }
        
        [Required]
        public int? CountryId { get; set; }
        
        [Required]
        public string? Name { get; set; }
    }

    public class CountryNameReqSearch : PagedReq
    {
        [Required]
        public int? CountryId { get; set; }
    }

}

File: admin/src/dtos/CountryName.ts
import Common from "../utility/Common";
import { CountryRes } from "./Country";
import { LanguageRes } from "./Language";
import { PagedReq } from "./PagedReq";

export interface CountryNameRes {
  countryNameId?: string;
  name?: string;

  languageId?: string;
  language?: LanguageRes;
  countryId?: string;
  country?: CountryRes;
}

export class CountryNameReqEdit {
  languageId?: string = "";
  countryId?: string = "";
  name?: string = "";

  constructor(languageId?: string, countryId?: string) {
    this.languageId = languageId;
    this.countryId = countryId;
  }
}

export class CountryNameReqSearch extends PagedReq {
  countryId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {countryId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.countryId = countryId;
  }
}